internal/ai/knowledgebase.go

```
@@ -0,0 +1,120 @@
+}
+
+// ReindexAll triggers a reconcile so snippets are re-embedded against the current model, e.g. after the embedding model changed.
+func (m *Manager) ReindexAll() {
+	go m.reconcile()
+}
+
+// snippetFingerprint signs the content and embedding model a snippet was last embedded against; any
+// change (edited content, switched model, changed dimensions) yields a new value and triggers reindex.
+func snippetFingerprint(item models.KnowledgeBaseItem, model string, dimensions int) string {
+	h := sha256.New()
+	fmt.Fprintf(h, "%s\x00%s\x00%s\x00%d", item.Title, item.Content, model, dimensions)
```

golangci-lint (2.12.2)

[error] Error return value of `fmt.Fprintf` is not checked

(errcheck)